pyphare/pyphare/pharesee/hierarchy/hierarchy.py

```
@@ -614,6 +613,50 @@ def dist_plot(self, **kwargs):
+            ret.time_hier[time][ilvl] = getattr(ufunc, method)(
+                *extract(time, ilvl), **kwargs
+            )
+    return ret
+
+
+def hierarchy_array_function(hier, func, types, args, kwargs):
+    def extract(time, ilvl):
+        return [x.level(ilvl, time) if type(x) is type(hier) else x for x in args]
+
+    time_hier = {}
+    for i, time in enumerate(hier.time_hier):
```

Ruff (0.15.9)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)